File: 01-week-1--starter-algorithms/03-day-4--selection-sort/javascript/selection_sort.js
function selectionSort(arr) {
  console.log(arr);

  const sortedValues = [];

  let i = arr.length;

  while (i > 0){
    const minValue = (Math.min(...arr))
    const index = arr.indexOf(minValue);
    console.log(minValue);
    i--;
  }

  return{

  }
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: [-1, 2, 3, 5]");
  console.log("=>", selectionSort([3, -1, 5, 2]));

  console.log("");

  // BENCHMARK HERE, and print the average runtime
  const longInput = [];

  for (let i = 0; i < 100; ++i) {
    longInput.push(Math.random());
  }
}

module.exports = selectionSort;

// Please add your pseudocode to this file
// And a written explanation of your solution

//find lowest value with Math.min(arr)
//remove lowest value from arr, and add it to sortedArr
//loop until all arr values are removed

File: 01-week-1--starter-algorithms/03-day-4--selection-sort/ruby/selection_sort.rb
def selection_sort(arr)
  # type your code in here
  sorted = []

  until arr.length == 0
    min = arr.length
    idx = arr.index(min)
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: [-1, 2, 3, 5]"
  print "=> "
  print selection_sort([3, -1, 5, 2])

  puts

  # Don't forget to add your own!

  # BENCHMARK HERE, you can print the average runtime
  long_input = []

  100.times { long_input << rand }
end

# find min value
# append min value to sorted array
# remove min value from original array
# iterate until complete
